sklearn/linear_model/stochastic_gradient.py

```diff
@@ -68,8 +71,29 @@ def __init__(self, loss, penalty='l2', alpha=0.0001, C=1.0,
         self.warm_start = warm_start
         self.average = average
 
-        self._validate_params()
+        if n_iter is not None:
+            warnings.warn("n_iter parameter is deprecated in 0.19 and will be"
+                          " removed in 0.21. Use max_iter and tol instead.",
+                          DeprecationWarning)
+            self.max_iter = n_iter
+            self.tol = 0.
+
+        if tol is None:
+            warnings.warn("tol parameter has been added in 0.19."
+                          "Set it to a positive value to stop the method "
+                          "before maximum number of iteration. "
+                          "Set it to 0 to force the number of iteration. "
+                          "From 0.21, default tol value will be changed to "
+                          "1e-3.", FutureWarning)
+
+        if max_iter == 5:
```

Comment:
Why 5 and not ``None``? Also, we currently require the user to set both ``tol`` and ``max_iter`` to get rid of the warnings, right?
That's slightly annoying. What we could do but would be slightly magic is "if the user set tol or max_iter, set the other one to a reasonable value (the future default) by default". @GaelVaroquaux @jnothman what do you think?

I like for users to have an easy way to future-proof, warning-free code.